Assets/Editor/FactoryInfoTest.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NUnit.Framework;
+using UnityEngine;
+using System.Xml.Serialization;
+using System.IO;
+
+public class FactoryInfoTest
+{
+    [Test]
+    public void TestSerialization()
+    {
+        FactoryInfo fi = new FactoryInfo();        
+        fi.PossibleProductions = new System.Collections.Generic.List<FactoryInfo.ProductionChain>();
+        //fi.InputSlots = new List<FactoryInfo.IntVector2>() { new FactoryInfo.IntVector2(1, 0) };
+        //fi.OutputSlots = new List<FactoryInfo.IntVector2>() { new FactoryInfo.IntVector2(2, 0) };
+        var chain1 = new FactoryInfo.ProductionChain() { Name="Iron smelting", ProcessingTime = 4.0f };
```

Comment:
This test file doesn't look formatted according to the rules in the CONTRIBUTING.md file. It's pretty hard to read.